Automattic/wp-calypso
# Hosting Command Palette: Add tests for useCommandPalette hook

Related to https://github.com/Automattic/dotcom-forge/issues/4952

## Proposed Changes

This PR adds tests for the `useCommandPalette` hook. It specifically adds tests for the sorting of the top-level commands (not the sites e.g. nested commands). Some test cases that the tests target:

* if there is no context specified and no `View my sites` command, the commands are expected to be returned in the order that they are added to an array of commands;
* if there is no context and `View my sites` command is present, the `View my sites` command is expected to be returned first followed by the other commands in the order that they were added to an array of commands;
* if the context is set to `/sites`, the `View my sites` command should be hidden and the commands should be returned in the order that they are added to the commands array;
* if the context is set to a custom context e.g `/settings`, the commands that match the context should be returned first followed by all other commands in the order that they were added to the commands array.

This setup should cover most of the logic for the top commands sorting.

## Testing Instructions


* Pull the changes from this PR to a local branch
* Run `yarn test-client -- client/components/command-palette/test/use-command-palette.tsx` in the terminal 
* Confirm that the tests are passing

## Pre-merge Checklist
